lib/saluki-components/src/sources/checks/builder/python.rs

```diff
@@ -0,0 +1,276 @@
+
+    fn initialized(&self) -> bool {
+        self.agent_check_class.is_some()
+    }
+
+    fn initialize() -> Self {
+        pyo3::prepare_freethreaded_python();
+        let result = Python::with_gil(|py| -> Result<PyObject, GenericError> {
+            let sys_path_attr = py
+                .import("sys")
+                .error_context("Could not import sys module.")
+                .and_then(|sys| sys.getattr("path").error_context("Could not get 'sys.path' attribute."))?;
+            let sys_path = sys_path_attr
+                .downcast::<PyList>()
+                .map_err(|_| GenericError::msg("Could not downcast 'sys.path' to list."))?;
+            // Add additional paths to sys.path to support loading checks.
+            let dist_path = Path::new("./dist");
+            let checks_d_path = dist_path.join("checks.d");
+            sys_path.insert(0, dist_path.to_string_lossy().as_ref()).unwrap(); // common modules are shipped in the dist path directly or under the "checks/" sub-dir
+            sys_path.insert(0, checks_d_path.to_string_lossy().as_ref()).unwrap(); // integrations-core legacy checks
+            sys_path.insert(0, "/etc/datadog-agent/checks.d/").unwrap(); // Agent checks folder
+            debug!("Updated Python system path (sys.path) to {:?}.", sys_path);
+            // Import the Datadog Checks module, ensuring it loads correctly, and grab a reference to the base AgentCheck class.
+            let dd_checks_module = match py.import("datadog_checks.checks") {
+                Ok(m) => m,
+                Err(e) => {
+                    if let Some(traceback) = e.traceback(py) {
+                        error!("Traceback: {}", traceback.format().expect("Could format traceback"));
+                    }
+                    return Err(generic_error!("Could not import datadog_checks module {:?}", e));
+                }
+            };
+            dd_checks_module
+                .getattr("AgentCheck")
+                .map(|c| c.unbind())
+                .error_context("Could not get AgentCheck class from datadog_checks module.")
+        });
+        match result {
+            Ok(agent_check_class) => {
+                info!("Python runtime loaded successfully and initialized for checks.");
+                Self {
+                    agent_check_class: Some(agent_check_class),
+                }
+            }
+            Err(e) => {
+                warn!(error = %e, "Failed to load/initialize Python runtime for checks. Python checks will be unavailable.");
+                Self {
+                    agent_check_class: None,
+                }
+            }
+        }
+    }
+}
+
+pub struct PythonCheckBuilder;
+
+impl CheckBuilder for PythonCheckBuilder {
+    fn build_check(
+        &self, name: &str, instance: &Instance, init_config: &Data, source: &MetaString,
+    ) -> Option<Arc<dyn Check + Send + Sync>> {
+        let env = PythonEnvBuilder::get_instance();
+        if env.initialized() {
```

Comment:
We generally prefer to have preconditions early on to let us return early. For example, if we know the Python environment isn't initialized, we should return `None` early:

```rust
let env = PythonEnvBuilder::get_instance();
if !env.initialized() {
    return None;
}

let agent_check_class = ...
```